src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -27,10 +27,24 @@
 
 #include "tusb_option.h"
 
-#if TUSB_OPT_DEVICE_ENABLED && CFG_TUSB_MCU == OPT_MCU_STM32F4
+#if TUSB_OPT_DEVICE_ENABLED && ( CFG_TUSB_MCU == OPT_MCU_STM32F4 || \
+                                 CFG_TUSB_MCU == OPT_MCU_STM32H7 )
+
+#if CFG_TUSB_MCU == OPT_MCU_STM32F4
+  #include "stm32f4xx.h"
+  // TODO Merge with OTG_HS
+  // Max endpoints for each direction
+  #define EP_MAX        USB_OTG_FS_MAX_IN_ENDPOINTS
+  #define EP_FIFO_SIZE  USB_OTG_FS_TOTAL_FIFO_SIZE
+#elif CFG_TUSB_MCU == OPT_MCU_STM32H7
+  #include "stm32h7xx.h"
+  // TODO There is no equivalent macro for max endpoints in H7 header.
```

Comment:
Maybe use `defined(USB_OTG_EPNUM_3) && !defined(USB_OTG_EPNUM_4)` (to know that there are 4 OUT endpoints), and then add the OUT and the IN counts? This same logic applies for the F1.

Reply:
oops, sorry, I followed the email link and merged the PR without knowing you guys are discussing . @pigrew I like your #ifdef EP checks. Probably will do that when doing more test/code with this dcd. Or if you could, submit your PR :wink:

Reply:
@pigrew I just checked, seem like all the header of st, only define up to USB_OTG_EPNUM_3, there is no USB_OTG_EPNUM_4 even on F7 which should support up to 6 !!! It seems to be only an bit definition only for the registers not the number of supported endpoints. I guess we have to it by hands :)